src/api/app/models/user.rb

```diff
@@ -75,6 +75,10 @@ class User < ApplicationRecord
   scope :in_beta, -> { where(in_beta: true) }
   scope :in_rollout, -> { where(in_rollout: true) }
 
+  scope :watchlisted_requests, ->(user) { BsRequest.where(id: user.watched_items.where(watchable_type: 'BsRequest').pluck(:watchable_id)).order('number DESC') }
+  scope :watchlisted_packages, ->(user) { Package.where(id: user.watched_items.where(watchable_type: 'Package').pluck(:watchable_id)).order('LOWER(name), name') }
+  scope :watchlisted_projects, ->(user) { Project.where(id: user.watched_items.where(watchable_type: 'Project').pluck(:watchable_id)).order('LOWER(name), name') }
```

Comment:
```suggestion
  scope :watched_requests, -> { BsRequest.where(id: watched_items.where(watchable_type: 'BsRequest').pluck(:watchable_id)).order('number DESC') }
  scope :watched_packages, -> { Package.where(id: watched_items.where(watchable_type: 'Package').pluck(:watchable_id)).order('LOWER(name), name') }
  scope :watched_projects, -> { Project.where(id: watched_items.where(watchable_type: 'Project').pluck(:watchable_id)).order('LOWER(name), name') }
```

Reply:
Scopes are probably not the best idea to fetch these records. Instead I've added instance methods because these are user specific querirs